src/dotnet/APIView/APIViewWeb/Managers/CopilotCommentsManager.cs

```diff
@@ -0,0 +1,186 @@
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace APIViewWeb.Managers
+{
+    public class CopilotCommentsManager : ICopilotCommentsManager
+    {
+        private readonly ICopilotCommentsRepository _copilotCommentsRepository;
+        private readonly IConfiguration _configuration;
+        private readonly OpenAIClient _openAIClient;
+
+        public CopilotCommentsManager(
+            ICopilotCommentsRepository copilotCommentsRepository,
+            IConfiguration configuration
+            )
+        {
+            _copilotCommentsRepository = copilotCommentsRepository;
+            _configuration = configuration;
+
+            _openAIClient = new OpenAIClient(
+                new Uri(_configuration["OpenAI:Endpoint"]),
+                new AzureKeyCredential(_configuration["OpenAI:Key"]));
+        }
+
+        public async Task<string> CreateDocumentAsync(string user, string badCode, string goodCode, string language, string comment, string[] guidelineIds)
+        {
+            var embedding = await GetEmbeddingsAsync(badCode); 
+
+            var document = new CopilotCommentModel();
+            document.BadCode = badCode;
+            document.GoodCode = goodCode;
+            document.Embedding = embedding;
+            document.Language = language;
+            document.Comment = comment;
+            document.GuidelineIds = guidelineIds;
+            document.ModifiedOn = DateTime.UtcNow;
+            document.ModifiedBy = user;
+
+            return await _copilotCommentsRepository.InsertDocumentAsync(document);
+        }
+
+        public async Task<UpdateResult> UpdateDocumentAsync(string user, string id, string badCode, string goodCode, string language, string comment, string[] guidelineIds)
+        {
+            var filter = GetIdFilter(id);
+
+            var updateBuilder = Builders<CopilotCommentModel>.Update;
+            var update = updateBuilder
+                .Set("ModifiedOn", DateTime.UtcNow)
+                .Set("ModifiedBy", user);
+
+            if (goodCode != null)
+            {
+                update = update.Set("GoodCode", goodCode);
+            }
+
+            if (language != null)
+            {
+                update = update.Set("Language", language);
+            }
+
+            if (comment != null)
+            {
+                update = update.Set("Comment", comment);
+            }
+
+            if (guidelineIds != null)
+            {
+                update = update.Set("GuidelineIds", guidelineIds);
+            }
+
+            if (badCode != null)
+            {
+                var embedding = await GetEmbeddingsAsync(badCode);
+                update = update
+                    .Set("BadCode", badCode)
+                    .Set("Embedding", embedding);
+            }
+
+            return await _copilotCommentsRepository.UpdateDocumentAsync(filter, update);
+        }
+
+        public async Task<UpdateResult> DeleteDocumentAsync(string user, string id)
```

Comment:
Nit, since this is just returning a task with no async, `await` and `async` shouldn't be required here.